M2/Macaulay2/packages/Msolve.m2

```diff
@@ -183,9 +183,9 @@ msolveRealSolutions(Ideal):=opts->(I)->(
     if solsp_0>0 then (print "Input ideal not zero dimensional, no solutions found."; return 0;);
     if (solsp_1)_0>1 then (print "unexpected msolve output, returning full output"; return solsp;);
     sols:=(solsp_1)_1;
-    if opts.Output=="rationalInterval" then return sols;
-    if opts.Output=="floatInterval" then return (1.0*sols);
-    if opts.Output=="float" then return (for s in sols list(for s1 in s list sum(s1)/2.0));
+    if opts#"output type"=="rationalInterval" then return sols;
+    if opts#"output type"=="floatInterval" then return (1.0*sols);
+    if opts#"output type"=="float" then return (for s in sols list(for s1 in s list sum(s1)/2.0));
```

Comment:
I'm guessing "float"=`RR` and "floatInterval"=`RRi`, but what is "rationalInterval"? Also, isn't it easier for the user to do `1.0*sols` if they want, and use `midpoint` for the "float" option?

Here are a few alternatives that I think would be more idiomatic here:
1. split this function into `msolveRealSolution`, `msolveRealIntervalSolution`, and `msolveRationalSolutionInterval` (if this exists in M2)
2. have this function always return `sols`, i.e. "rationalInterval" option, but have helper functions like `midpoint` for turning that into a float interval or float, which could also be used in other places idiomatically
3. add a new input of type `Ring` to this function, which could be `QQ` or `RR` or `RRi` and cast to that ring.

Reply:
I don't have a strong feeling here, but mostly the interval return might be a bit hard for people to read. But it makes sense for the default as it is still the most accurate answer. 

I don't like suggestion 3 so much as expect feeding anything other than QQ or a finite field into msolve is going to create infinite problems.

Reply:
I agree that the default should be the most accurate, i.e. `sols`. Why is it not returning RRi interval objects?

For option 3, it should definitely throw an error if the ring is anything other than the handful of fields we can handle.

Reply:
I am not returning RRi objects since mostly since I am not familiar with these. 

The default return is what msolve would return, unedited. So it is intervals of rationals. 

I have left this as is in my most recent update, but if others want to change it that is also fine with me.